test/test.js

```diff
@@ -650,6 +651,85 @@ const createTestSuite = (useElectronNet) => {
       })
     })
 
+    it('should handle aborts before request', function () {
+      this.timeout(500)
+      const abort = new AbortController()
+      abort.abort()
+      url = `${base}timeout`
+      opts = {
+        useElectronNet,
+        signal: abort.signal
+      }
+      return expect(fetch(url, opts)).to.eventually.be.rejected
+        .and.be.an.instanceOf(FetchError)
+        .and.have.property('type', 'abort')
+    })
+
+    it('should handle aborts during a request', function () {
+      this.timeout(500)
+      const abort = new AbortController()
+      setTimeout(() => {
+        abort.abort()
+      }, 100)
+      url = `${base}timeout`
+      opts = {
+        useElectronNet,
+        signal: abort.signal
+      }
+      return expect(fetch(url, opts)).to.eventually.be.rejected
+        .and.be.an.instanceOf(FetchError)
+        .and.have.property('type', 'abort')
+    })
+
+    it('should handle aborts during a response', function () {
+      this.timeout(500)
+      const abort = new AbortController()
+      setTimeout(() => {
+        abort.abort()
+      }, 100)
+      url = `${base}slow`
+      opts = {
+        useElectronNet,
+        signal: abort.signal
+      }
+      return fetch(url, opts).then(res => {
+        expect(res.ok).to.be.true
+        return expect(res.text()).to.eventually.be.rejectedWith(FetchError)
+          .and.to.satisfy(e => e.message.endsWith('request aborted'))
```

Comment:
Why have a different check here, compared to the previous test ? (which checks `        .and.have.property('type', 'abort')`)

Reply:
Answer to myself : the abort error on the response stream is caught by the `.text()` code and re-thrown

Reply:
Yeah, I wasn't really sure how to properly test if stream emits `error`, and `text()` (or anything using `consumeBody`) would just rethrow any `error` that is emitted by stream before its end.